pkg/planner/core/logical_plan_builder.go

```diff
@@ -2917,13 +2917,36 @@ func (g *gbyResolver) Leave(inNode ast.Node) (ast.Node, bool) {
 			return inNode, false
 		}
 	case *ast.PositionExpr:
-		pos, isNull, err := expression.PosFromPositionExpr(g.ctx.GetExprCtx(), v)
+		pos, isNull, isParamMarkerExpr, err := expression.PosFromPositionExpr(g.ctx.GetExprCtx(), v)
 		if err != nil {
 			g.err = plannererrors.ErrUnknown.GenWithStackByArgs()
 		}
 		if err != nil || isNull {
 			return inNode, false
 		}
+		if pos == 0 && isParamMarkerExpr {
+			// In Spring ORM, sometimes such prepare statements are generated.
+			// prepare stmt from 'select name, ? from t1 group by name, ? order by name';
+			// set @a="0", @b="0";
+			// Grouping by 0 is illegal, but here it should be of string type.
+			// In fact, any string is valid here.
+			// So, we need to try to find the ParamMarkerExpr in the SELECT field and give its index to pos.
+			if len(g.fields) == 1 {
+				// for example:
+				// prepare stmt from 'select name from t1 group by name, ? order by name';
+				// set a=0;
+				// execute stmt using @b;
+				pos = 1
```

Comment:
`group by "0"` is equal to `group by 1`, this behavior is really weird, please comment that this behavior is from MySQL.